### ISSUE TITLE
TypeError: fs.existsSync is not a function

### ISSUE THREAD

**Issue by ufoscout** [SitePen/remap-istanbul#21]
Hi all,

we are integrating remap-istanbul in our workflow to have coverage report on our ts files; however, when executed it always throws this exception:
```
myApp\node_modules\remap-istanbul\lib\remap.js:139
                        if (!fs.existsSync(filePath)) {
                                ^

TypeError: fs.existsSync is not a function
    at readFile (myApp\node_modules\remap-istanbul\lib\remap.js:139:12)
    at myApp\node_modules\remap-istanbul\lib\remap.js:155:18
    at Array.forEach (native)
    at myApp\node_modules\remap-istanbul\lib\remap.js:153:22
    at Array.forEach (native)
    at remap (myApp\node_modules\remap-istanbul\lib\remap.js:152:12)
    at DestroyableTransform._transform (myApp\node_modules\remap-istanbul\lib\gulpRemapIstanbul.js:31:20)
    at DestroyableTransform.Transform._read (myApp\node_modules\readable-stream\lib\_stream_transform.js:172:10)
    at DestroyableTransform.Transform._write (myApp\node_modules\readable-stream\lib\_stream_transform.js:160:12)
    at doWrite (myApp\node_modules\readable-stream\lib\_stream_writable.js:333:12)
    at writeOrBuffer (myApp\node_modules\readable-stream\lib\_stream_writable.js:319:5)
    at DestroyableTransform.Writable.write (myApp\node_modules\readable-stream\lib\_stream_writable.js:246:11)
    at write (myApp\node_modules\vinyl-fs\node_modules\readable-stream\lib\_stream_readable.js:623:24)
    at flow (myApp\node_modules\vinyl-fs\node_modules\readable-stream\lib\_stream_readable.js:632:7)
    at DestroyableTransform.pipeOnReadable (myApp\node_modules\vinyl-fs\node_modules\readable-stream\lib\_stream_readable.js:664:5)
    at emitNone (events.js:67:13)
```

We lauch it with gulp:
```
var remapIstanbul = require('remap-istanbul/lib/gulpRemapIstanbul');

gulp.task('coverage', ['test'], function () {
  return gulp.src('test/coverage/coverage.json')
    .pipe(remapIstanbul({
      reports: {
        'text' : 'test/coverage-ts/coverage.txt',
        'lcovonly' : 'test/coverage-ts/lcov.dat',
        'json': 'test/coverage-ts/coverage.json',
        'html': 'test/coverage-ts/html-report'
      },
      basePath : './app/'
    }));
    //.pipe(remapIstanbul())
    //.pipe(gulp.dest('test/coverage-ts'));
});
```

We tried with nodejs 4.1.0, 4.2.2 and 5.1.0.
The version of remap-istanbul is 0.4.0.

With the following change in the remap.js everything works as expected:
```
var nodeFS = require('fs'); // ADD THIS LINE HERE

if (typeof define !== 'function') { /* istanbul ignore next */ var define = require('amdefine')(module); }
define([
	'require',
	'exports',
	'./node!istanbul/lib/collector',
	'./node!path',
	'./node!fs',
	'./node!source-map/lib/source-map-consumer'
], function (require, exports, Collector, path, fs, smc) {
	/* global WeakMap */

  fs = nodeFS;  // ADD THIS LINE HERE
  ...
```

**Comment by kitsonk** [comment]
Hmmm... I suspect it is related to #19.

Are you using Node 5 and npm 3?  If so, in the project you are using, can you provide a directory listing of your `node_modules`?

**Comment by ufoscout** [comment]
Here the complete list of subdirectories in the node_modules directory:

```
abbrev
accepts
acorn
adm-zip
after
align-text
alter
amdefine
ansi
ansi-cyan
ansi-green
ansi-red
ansi-regex
ansi-styles
ansi-wrap
anymatch
archy
are-we-there-yet
argh
argparse
arr-diff
arr-flatten
arr-union
array-differ
array-index
array-slice
array-union
array-uniq
array-unique
arraybuffer.slice
arrify
asn1
assert-plus
assertion-error
async
async-each
async-each-series
async-foreach
autoprefixer-core
aws-sign2
backo2
balanced-match
base64-arraybuffer
base64id
batch
beeper
benchmark
better-assert
binary-extensions
binaryextensions
bindings
bl
blob
block-stream
bluebird
body-parser
boom
bower-config
brace-expansion
braces
browser-sync
browser-sync-client
browser-sync-spa
browser-sync-ui
browserify-zlib
browserslist
bufferstreams
bufferutil
builtin-modules
bytes
callsite
camelcase
camelcase-keys
caniuse-db
caseless
center-align
chalk
chokidar
clap
cli
cli-color
cliui
clone
clone-stats
code-point-at
color
color-convert
color-name
color-string
colornames
colors
colorspace
combined-stream
commander
component-bind
component-emitter
component-inherit
concat-map
concat-stream
concat-with-sourcemaps
config-chain
configstore
connect
connect-history-api-fallback
connect-modrewrite
console-browserify
content-type
convert-source-map
cookie
core-js
core-util-is
cross-spawn
cross-spawn-async
cryptiles
csso
ctype
custom-event
d
date-now
dateformat
deap
debug
decamelize
deep-extend
deep-freeze
deep-is
defaults
definition-header
del
delayed-stream
delegates
depd
deprecated
destroy
detect-indent
dev-ip
di
diagnostics
diff
dom-serialize
dom-serializer
domelementtype
domhandler
domutils
dot-object
duplexer
duplexer2
duplexify
easy-extender
eazy-logger
ee-first
emits
emitter-steward
enabled
end-of-stream
engine.io
engine.io-client
engine.io-parser
ent
entities
env-variable
error-ex
es5-ext
es6-iterator
es6-promise
es6-symbol
es6-weak-map
escape-html
escape-string-regexp
escodegen
esprima
estraverse
esutils
etag
event-emitter
event-stream
eventemitter3
exit
expand-braces
expand-brackets
expand-range
express
extend
extend-shallow
extglob
extsprintf
fancy-log
fast-levenshtein
filelist.txt
filename-regex
fileset
fill-range
finalhandler
find-index
find-up
findup-sync
first-chunk-stream
flagged-respawn
for-in
for-own
foreach
forever-agent
fork-stream
form-data
formidable
foxy
fresh
from
fs
fs-extra
fstream
gauge
gaze
generate-function
generate-object-property
get-stdin
glob
glob-base
glob-parent
glob-stream
glob-watcher
glob2base
global
globby
globule
glogg
got
graceful-fs
graceful-readlink
gulp
gulp-add-src
gulp-angular-templatecache
gulp-autoprefixer
gulp-bump
gulp-concat
gulp-concat-sourcemap
gulp-csso
gulp-filter
gulp-flatten
gulp-folders
gulp-footer
gulp-header
gulp-if
gulp-inject
gulp-jshint
gulp-load-plugins
gulp-match
gulp-minify-html
gulp-ng-annotate
gulp-remove-lines
gulp-rename
gulp-replace
gulp-rev
gulp-rev-replace
gulp-sass
gulp-shell
gulp-size
gulp-sourcemaps
gulp-strip-code
gulp-tslint
gulp-typescript
gulp-uglify
gulp-usemin
gulp-useref
gulp-util
gulplog
gzip-size
handlebars
har-validator
has-ansi
has-binary
has-binary-data
has-cors
has-flag
has-gulplog
has-unicode
hawk
hoek
hosted-git-info
htmlparser2
http-errors
http-proxy
http-signature
iconv-lite
immutable
indent-string
indexof
infinity-agent
inflight
inherits
ini
interpret
invert-kv
is-absolute
is-absolute-url
is-arrayish
is-binary-path
is-buffer
is-builtin-module
is-dotfile
is-equal-shallow
is-extendable
is-extglob
is-finite
is-fullwidth-code-point
is-glob
is-my-json-valid
is-npm
is-number
is-path-cwd
is-path-in-cwd
is-path-inside
is-primitive
is-property
is-redirect
is-relative
is-relative-url
is-stream
is-utf8
is-valid-glob
isarray
isemail
isobject
isstream
istanbul
istextorbinary
jasmine-core
joi
joi-assert
js-base64
js-string-escape
js-yaml
jsesc
jshint
jshint-stylish
json-pointer
json-stringify-safe
json3
jsonfile
jsonpointer
karma
karma-coverage
karma-jasmine
karma-ng-html2js-preprocessor
karma-phantomjs-launcher
kew
kind-of
kuler
latest-version
lazy-cache
lazy.js
lcid
levn
liftoff
limiter
load-json-file
localtunnel
lodash
lodash.assign
lodash.defaults
lodash.escape
lodash.isarguments
lodash.isarray
lodash.isobject
lodash.isplainobject
lodash.istypedarray
lodash.keys
lodash.keysin
lodash.merge
lodash.pad
lodash.padleft
lodash.padright
lodash.repeat
lodash.restparam
lodash.template
lodash.templatesettings
lodash.toplainobject
lodash.values
lodash._arraycopy
lodash._arrayeach
lodash._baseassign
lodash._basecopy
lodash._basefor
lodash._basetostring
lodash._basevalues
lodash._bindcallback
lodash._createassigner
lodash._createpadding
lodash._escapehtmlchar
lodash._escapestringchar
lodash._getnative
lodash._htmlescapes
lodash._isiterateecall
lodash._isnative
lodash._objecttypes
lodash._reescape
lodash._reevaluate
lodash._reinterpolate
lodash._reunescapedhtml
lodash._shimkeys
log-symbols
log4js
longest
loud-rejection
lowercase-keys
lru-cache
lru-queue
map-obj
map-stream
media-typer
memoizee
meow
merge
merge-stream
micromatch
mime
mime-db
mime-types
minichain
minimatch
minimist
minimize
ministyle
minitable
miniwrite
mkdirp
modify-filename
moment
mout
ms
multimatch
multipipe
nan
negotiator
nested-error-stacks
next-tick
ng-annotate
node-gyp
node-sass
node-useref
node-uuid
nopt
normalize-package-data
normalize-path
npmconf
npmlog
num2fraction
number-is-nan
oauth-sign
object-assign
object-component
object-keys
object-path
object.omit
on-finished
once
open
openurl
opn
opt-merger
optimist
optionator
options
orchestrator
ordered-ast-traverse
ordered-esprima-props
ordered-read-streams
os-homedir
os-locale
os-shim
os-tmpdir
osenv
package-json
pad-left
pako
parse-glob
parse-json
parsejson
parseqs
parseuri
parseurl
parsimmon
path
path-array
path-exists
path-is-absolute
path-is-inside
path-type
pause-stream
phantomjs
pify
pinkie
pinkie-promise
pjs
plexer
plugin-error
plugin-log
plur
portscanner
postcss
prelude-ls
prepend-http
preserve
pretty-bytes
pretty-hrtime
process
process-nextick-args
progress
propprop
proto-list
proxy-middleware
qs
query-string
randomatic
range-parser
raw-body
rc
rcfinder
rcloader
read-all-stream
read-pkg
read-pkg-up
readable-stream
readdirp
rechoir
redent
regex-cache
registry-url
remap-istanbul
repeat-element
repeat-string
repeating
replace-ext
replacestream
request
request-progress
require-dir
requires-port
resolve
resp-modifier
rev-hash
rev-path
right-align
rimraf
sass-graph
semver
semver-diff
send
sequencify
serve-index
serve-static
shelljs
sigmund
signal-exit
simple-fmt
simple-is
sntp
socket.io
socket.io-adapter
socket.io-client
socket.io-parser
source-map
sparkles
spawn-sync
spdx-correct
spdx-exceptions
spdx-expression-parse
spdx-license-ids
split
sprintf-js
stable
statuses
stream-combiner
stream-consume
stream-throttle
streamqueue
strict-uri-encode
string-length
string-width
stringmap
stringset
stringstream
string_decoder
strip-ansi
strip-bom
strip-bom-stream
strip-indent
strip-json-comments
success-symbol
supports-color
tar
ternary-stream
text-hex
text-table
textextensions
tfunk
throttleit
through
through2
through2-filter
tildify
timed-out
timers-ext
to-absolute-glob
to-array
topo
tough-cookie
trim-newlines
tryor
tsd
tslint
tunnel-agent
type-check
type-detect
type-is
typedarray
typescript
ua-parser-js
ucfirst
uglify-js
uglify-save-license
uglify-to-browserify
uid-number
ultron
underscore
underscore.string
unique-stream
universal-analytics
unpipe
update-notifier
uri-templates
user-home
useragent
utf-8-validate
utf8
util
util-deprecate
utils-merge
uuid
v8flags
validate-npm-package-license
verror
vinyl
vinyl-file
vinyl-fs
vinyl-sourcemaps-apply
void-elements
weinre
which
window-size
wiredep
wordwrap
wrap-ansi
wrappy
wrench
ws
xdg-basedir
xmlhttprequest
xregexp
xtend
y18n
yargs
```

**Comment by kitsonk** [comment]
Yes, thanks for that... there is a `fs` module that some dependency is loading and the way we load modules is getting confused because it is finding `fs` as a peer of itself! :-(

It is a dupe of #19.  I am really starting to hate npm.  Who the hell a) [puts built in modules in their dependency list](https://www.npmjs.com/browse/depended/fs) and b) allows people to [add shitty modules which conflict with the built in modules](https://www.npmjs.com/package/fs) to npm?

A temporary fix is to not use npm3, while we try to figure out how to make a permanent fix in #19.  I am going to close this though as a dupe of #19.

**Comment by nwawuikeobinna** [comment]
I came across that same error in one of my application, took quite some time to figure it out.. When one encounters such error the sure way is to chill and mull over the code once once more to check for typos or wrong importations..
